cli/pkg/schema/yaml_context.go

```diff
@@ -0,0 +1,96 @@
+package schema
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/charmbracelet/lipgloss"
+	"github.com/nitrictech/nitric/cli/internal/style/colors"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+type YamlContextBuilder struct {
+	strings.Builder
+}
+
+func (b *YamlContextBuilder) WriteError(s string) (int, error) {
+	errStr := lipgloss.NewStyle().Foreground(colors.Red).Render(fmt.Sprintf("\t# <-- %s", s))
+	return b.Builder.WriteString(errStr)
+}
+
+func (b *YamlContextBuilder) WriteYamlKey(s interface{}, indent int) (int, error) {
+	indentStr := strings.Repeat("  ", indent)
+	// If the key is an integer, it means its an array index, replace it with a dash.
+	keyStr := ""
+	if _, err := strconv.Atoi(fmt.Sprintf("%v", s)); err == nil {
```

Comment:
does `s` come through as its original type I'm assuming it does based on the arg.

A switch with type might be more suited here:

```go
switch v := s.(type) {
	case int:
		// v is an int
	case string:
		// v is a string
	default:
		// v is :shrug:
}
```